framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -706,6 +707,18 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
     }
   }
 
+  std::unordered_set<Node *> msm_connected_nodes;
+
+  // Deleting elements may produce isolated nodes.
+  // Loops for identifying and removing such nodes from mortar segment mesh.
+  for (const auto & element : _mortar_segment_mesh->element_ptr_range())
+    for (auto & n : element->node_ref_range())
+      msm_connected_nodes.insert(&n);
+
+  for (const auto & node : _mortar_segment_mesh->node_ptr_range())
+    if (!msm_connected_nodes.count(node))
+      _mortar_segment_mesh->delete_node(node);
```

Comment:
Do these orphaned nodes have a practical impact on the computation? Or does this make a debug mesh or nodal geometry look weird if they're left in? Just curious if there's a motivation for this other than making the mesh "logically clean"

Reply:
I don't think they have an impact on the computation; it doesn't for the case I tried out. But those nodes still reside in the `mortar_segment_mesh`, which makes the exodus file (object) different from the case in which we do not wrap the sidesets/lowerds around the block. My main motivation was to ensure the `mortar_segment_mesh` is as close as possible to the reference situation with lower-d only on the contact patch, otherwise, for the bouncing block, e.g., the mesh has double the number of nodes compared to the reference case.

Reply:
Alright we can keep this then. If it shows up in profiling ever, we can consider removing it